File: Search-Insert-Position.c
 int searchInsert(int* nums, int numsSize, int target){
     int index=0;
    int check=0;

     for(int k=0;k<numsSize;k++ ) {
         if(  nums[k] >= target) {
             index = k;
             check =1;
              break;
         }
     }
         if( check == 1) 
            return index;
        else  
                 return numsSize;
                return NULL;

 }

File: java/Search-Insert-Position.java
https://leetcode.com/problems/search-insert-position

class Solution {
    public int searchInsert(int[] nums, int target) {

        int start = 0;
        int end = nums.length - 1;

        while( start <= end ) {
            // calculate the mid 
            int mid = start + ( end - start ) / 2;
                if( nums[mid] > target ) {
                    end = mid - 1;
                }
                else if( nums[mid] < target ) {
                    start = mid + 1;
                }
                else {
                    return mid;
                }
        }
        return start;
    }
}
